pinot-spi/src/main/java/org/apache/pinot/spi/config/table/ingestion/IngestionConfig.java

```diff
@@ -49,10 +52,19 @@ public class IngestionConfig extends BaseJsonConfig {
   private ComplexTypeConfig _complexTypeConfig;
 
   @JsonPropertyDescription("Config related to the SchemaConformingTransformer")
+  @JsonProperty("schemaConformingTransformerConfig")
   private SchemaConformingTransformerConfig _schemaConformingTransformerConfig;
 
-  @JsonPropertyDescription("Config related to the SchemaConformingTransformerV2")
-  private SchemaConformingTransformerV2Config _schemaConformingTransformerV2Config;
+  @JsonPropertyDescription("Config related to the SchemaConformingTransformerV2 (backward compatibility)")
+  @JsonProperty("schemaConformingTransformerV2Config")
+  private void setSchemaConformingTransformerV2Config(JsonNode schemaConformingTransformerV2Config) {
+    // Map _schemaConformingTransformerV2Config JSON into SchemaConformingTransformerConfig
+    if (schemaConformingTransformerV2Config != null && _schemaConformingTransformerConfig == null) {
+      ObjectMapper objectMapper = new ObjectMapper();
+      _schemaConformingTransformerConfig =
+          objectMapper.convertValue(schemaConformingTransformerV2Config, SchemaConformingTransformerConfig.class);
+    }
+  }
```

Comment:
Does it work the same if we simply put:
```suggestion
  public void setSchemaConformingTransformerV2Config(SchemaConformingTransformerConfig schemaConformingTransformerConfig) {
    _schemaConformingTransformerConfig = schemaConformingTransformerConfig;
  }
```

All we need to do is mapping both key to the same property right?

Reply:
> All we need to do is mapping both key to the same property right?

Yes, but would the suggested change work in the same way?
In which case, the `setSchemaConformingTransformerV2Config(SchemaConformingTransformerConfig schemaConformingTransformerConfig)` would be called? I thought when 
```
{"schemaConformingTransformerV2Config" : {...}}
```
appears in Json, it would internally trigger `setSchemaConformingTransformerV2Config(JsonNode schemaConformingTransformerV2Config)` only (JsonNode as type).

Reply:
Based on my understanding, it should be able to directly construct the `SchemaConformingTransformerConfig` properly. You can probably add a test for backward compatibility

Reply:
Good idea! Added the UT. Verified that this works. Thanks for the proposal.